zzz_generated/app-common/app-common/zap-generated/cluster-objects.h

```diff
@@ -0,0 +1,178 @@
+#pragma once
+
+#include <app/data-model/DecodableList.h>
+#include <app/data-model/Decode.h>
+#include <app/data-model/Encode.h>
+#include <app/data-model/List.h>
+#include <app/util/basic-types.h>
+#include <array>
+#include <lib/core/CHIPTLV.h>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace clusters {
+namespace TestCluster {
+constexpr ClusterId kClusterId = 0x0000050F;
+
+enum class SimpleEnum : uint8_t
+{
+    UNSPECIFIED = 0x00,
+    VALUEA      = 0x01,
+    VALUEB      = 0x02,
+    VALUEC      = 0x03
+};
+
+namespace TestListStructOctet {
```

Comment:
Please remind me again why this needs to be a namespace?  That is, why can't we have:
```
struct TestListStructOctet
{
  enum FieldId .... etc
}
```
without the nested `struct Type` complication?

Reply:
I should be able to use the field-ids alone without requiring the Type. I'm actually thinking of teasing apart this file into two - one that contains just the field-ids, and one that contains just the types.

More-over, I realize that that Decodable and non-Decodable types should live in the same namespace for that 'data model element'.

Reply:
Have shifted the Decodable and non-Decodable types under the same namespace now.